.github/workflows/run-e2e-tests.yml

```diff
@@ -605,13 +605,54 @@ jobs:
         shell: bash
         run: echo "workflow_id=$(uuidgen)" >> "$GITHUB_OUTPUT"
 
+  # Check if the used ECR registry is public or private
+  # Skip building image if public registry is used
+  check-registry-type:
+    name: Check used ECR registry type
+    needs: [validate-inputs]
+    if: ${{ needs.validate-inputs.outputs.aws_test_secrets_key }}
+    runs-on: ubuntu-latest
+    environment: integration
+    permissions:
+      id-token: write
+    outputs:
+      is-public-registry: ${{ steps.is-public-registry.outputs.is-public-registry}}
+    steps:
+      - name: Get test secrets from AWS Secret Manager
+        id: aws-test-secrets
+        uses: smartcontractkit/.github/actions/ctf-fetch-aws-secret@921f4b0ca850dd473dcef9082e3169ccbb83cc52 # ctf-fetch-aws-secret@0.0.0
+        with:
+          secret_id: ${{ needs.validate-inputs.outputs.aws_test_secrets_key }}
+          aws_region: ${{ secrets.QA_AWS_REGION }}
+          aws_role_to_assume: ${{ secrets.QA_AWS_ROLE_TO_ASSUME }}
+
+      - name: Parse and mask test secrets
+        id: parsed-aws-test-secrets
+        uses: smartcontractkit/.github/actions/parse-and-mask-test-secrets/0.1.0 # parse-and-mask-test-secrets@0.1.0
```

Comment:
`uses: smartcontractkit/.github/actions/parse-and-mask-test-secrets@parse-and-mask-test-secrets/0.1.0`